pkg/maps/nodemap/node_map_v2.go

```diff
@@ -145,6 +149,76 @@ func LoadNodeMapV2() (MapV2, error) {
 	return &nodeMapV2{bpfMap: bpfMap}, nil
 }
 
+// migrateV1 will migrate the v1 NodeMap to this NodeMapv2
+//
+// Ensure this always occurs BEFORE we begin handling K8s Node events or else
+// both this migration and the node events will be writing to the map.
+//
+// This migration will leave the v1 NodeMap preset on the filesystem.
+// This is due to some unfortunately versioning requirements which forced this
+// migration to occur within a patch release.
+//
+// When Cilium reaches v1.17 the v1 NodeMap will no longer be required and we
+// can unpin the map after migration.
+func (m *nodeMapV2) migrateV1(NodeMapName string, EncryptMapName string) error {
+	log.Debug("Detecting V1 to V2 migration")
+
+	// load v1 node map
+	nodeMapPath := bpf.MapPath(NodeMapName)
+	v1, err := ebpf.LoadPinnedMap(nodeMapPath)
+	if errors.Is(err, unix.ENOENT) {
+		log.Debug("No v1 node map found, skipping migration")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	nodeMap := nodeMap{
+		bpfMap: v1,
+	}
+
+	// load encrypt map to get current SPI
+	encryptMapPath := bpf.MapPath(EncryptMapName)
+	en, err := ebpf.LoadPinnedMap(encryptMapPath)
+	if errors.Is(err, unix.ENOENT) {
+		log.Debug("No encrypt map found, skipping migration")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	defer en.Close()
+
+	var SPI uint8
+	if err = en.Lookup(uint32(0), &SPI); err != nil {
+		return err
+	}
+
+	// reads v1 map entries and writes them to V2 with the latest SPI found
+	// from EncryptMap
+	parse := func(k *NodeKey, v *NodeValue) {
+		v2 := NodeValueV2{
+			NodeID: v.NodeID,
+			SPI:    SPI,
+		}
+
+		log.WithFields(logrus.Fields{
+			logfields.NodeID: v2.NodeID,
+			logfields.IPAddr: k.IP,
+			logfields.SPI:    v2.SPI,
+		}).Debug("Migrating V1 node map entry to V2")
```

Comment:
Not opposed to reducing it tho, wonder if seeing each specific value is worth it. 

Its true tho, itll blow up the output when theres 40+ nodes. 

Okay i think ill reduce it.

Reply:
Resolved in latest push.

Reply:
Thanks! We may occasionally enable debug mode in clusters with hundreds or even thousands of nodes so that's why I thought it was probably worth reducing.